### PR TITLE
[v4] v4l2-request-test.c: introduce parsing for 'long options'

### PR DESCRIPTION
**Pull request by rzerres**
- enable getopt_long supporting long option style
  (e.g --video-path, --fps, etc.)
- try to be consistent which 'v4l2-compliance' binary when
  choosing option names
- adapt help message
- to adopt the slice-path you need to explicitly prepend an
  argument-option (-s or --slice-path) with the path as an option-string
- adopt summary output to be in line with long options
- typo corrections
- update code style (tabify vs whitespace, alingement)

Signed-off-by: Ralf Zerres <ralf.zerres@networkx.de>

### CODE REVIEW

File: v4l2-request-test.c [hunk 1]
```diff
@@ -279,6 +307,10 @@ int main(int argc, char *argv[])
 			config.drm_driver = strdup(optarg);
 			break;
 		case 's':
+			free(config.slices_path);
```
Comment (paulkocialkowski) [inline v4l2-request-test.c]: We didn't have any config.slices_path allocated at this point, did we? Unless I'm mistaken, this should be removed.

File: v4l2-request-test.c [hunk 2]
```diff
@@ -257,7 +266,26 @@ int main(int argc, char *argv[])
 	setup_config(&config);
 
 	while (1) {
-		opt = getopt(argc, argv, "v:m:d:D:s:f:P:ilqh");
+	        int option_index = 0;
+		static struct option long_options[] = {
```
Comment (paulkocialkowski) [inline v4l2-request-test.c]: Please fix the following block's alignment to one 3 tabs (1 more than where the declaration starts).

File: v4l2-request-test.c [hunk 3]
```diff
@@ -257,7 +266,26 @@ int main(int argc, char *argv[])
...
+						       {"device",        required_argument, 0, 'v' },
```
Comment (paulkocialkowski) [inline v4l2-request-test.c]: Please add a whitespace between { and the first string in each declaration.

File: v4l2-request-test.c [hunk 4]
```diff
@@ -64,42 +64,51 @@ struct format_description formats[] = {
...
-	printf(" Video path: %s\n", config->video_path);
-	printf(" Media path: %s\n", config->media_path);
-	printf(" DRM path: %s\n", config->drm_path);
-	printf(" DRM driver: %s\n", config->drm_driver);
-	printf(" Slices path: %s\n", config->slices_path);
-	printf(" Slices filename format: %s\n", config->slices_filename_format);
-	printf(" FPS: %d\n\n", config->fps);
+	printf(" Video device:  %s\n", config->video_path);
+	printf(" Media device:  %s\n", config->media_path);
+	printf(" DRM device:    %s\n", config->drm_path);
```
Comment (paulkocialkowski) [inline v4l2-request-test.c]: Please align with tabs instead of whitespaces.

File: v4l2-request-test.c [hunk 5]
```diff
@@ -64,42 +64,51 @@ struct format_description formats[] = {
...
+	printf(" Frames p/sec:  %d\n\n", config->fps);
 
 	printf("Preset:\n");
-	printf(" Name: %s\n", preset->name);
-	printf(" Description: %s\n", preset->description);
-	printf(" License: %s\n", preset->license);
-	printf(" Attribution: %s\n", preset->attribution);
-	printf(" Width: %d\n", preset->width);
-	printf(" Height: %d\n", preset->height);
+	printf(" Name:         %s\n", preset->name);
```
Comment (paulkocialkowski) [inline v4l2-request-test.c]: Please use tabs for alignment.

File: v4l2-request-test.c [hunk 6]
```diff
@@ -64,42 +64,51 @@ struct format_description formats[] = {
...
-	printf(" Slices path: %s\n", config->slices_path);
-	printf(" Slices filename format: %s\n", config->slices_filename_format);
-	printf(" FPS: %d\n\n", config->fps);
+	printf(" Video device:  %s\n", config->video_path);
+	printf(" Media device:  %s\n", config->media_path);
+	printf(" DRM device:    %s\n", config->drm_path);
+	printf(" DRM driver:    %s\n", config->drm_driver);
+	printf(" Slices path:   %s\n", config->slices_path);
+	printf(" Slices format: %s\n", config->slices_filename_format);
+	printf(" Frames p/sec:  %d\n\n", config->fps);
```
Comment (paulkocialkowski) [inline v4l2-request-test.c]: Please keep it called FPS to match the argument name.

File: v4l2-request-test.c [hunk 7]
```diff
@@ -64,42 +64,51 @@ struct format_description formats[] = {
...
-	       " -D [DRM driver]                DRM driver to use\n"
-	       " -s [slices filename format]    format for filenames in the slices path\n"
-	       " -f [fps]                       number of frames to display per second\n"
-	       " -P [video preset]              video preset to use\n"
-	       " -i                             enable interactive mode\n"
-	       " -l                             loop preset frames\n"
-	       " -q                             enable quiet mode\n"
-	       " -h                             help\n\n"
-	       "Video presets:\n");
+	       " -v, --device <dev>\n"
```
Comment (paulkocialkowski) [inline v4l2-request-test.c]: Could you try and keep the help text on the same line as the arguments?
Also, I don't really see the point of adding --device as a shorthand for --video-device but we can keep it if you think it's better that way.

File: v4l2-request-test.c [hunk 8]
```diff
@@ -306,6 +338,10 @@ int main(int argc, char *argv[])
 		case '?':
 			print_help();
 			goto error;
+		default:
+			printf("Unknown option (0%o)\n",
+			       opt);
```
Comment (paulkocialkowski) [inline v4l2-request-test.c]: I think we can just ignore unknown options since we already print out a summary.

**Review by paulkocialkowski — COMMENTED** [review]
Thanks for this new spin! See comments about the second patch. The first one can be dropped already.

File: v4l2-request-test.c (hunk 1 above)
Comment (rzerres) [reply v4l2-request-test.c]: If you are not going to go with the default (which will pick ./data/<preset-name>), and you are selecting -s the following argument will let to the path where the slice-data are stored. At least, this is the idea.
but right, the parameter isn't preset ....

File: v4l2-request-test.c (hunk 2 above)
Comment (rzerres) [reply v4l2-request-test.c]: if i got you right, I need to remove all whitespace and put one extra tab?

File: v4l2-request-test.c (hunk 6 above)
Comment (rzerres) [reply v4l2-request-test.c]: ok.

File: v4l2-request-test.c (hunk 7 above)
Comment (rzerres) [reply v4l2-request-test.c]: well, v4l2-compliance is using -d, --device for the video device. In v4l2-request-test -d goes with the drm device. That makes more sense to me. Having a correct destinction, --video-device is choosen. But admins/developers might be used to assume --device be the video device. Therfore both parameters are accecpted. I have no problem to remove --device, so.

File: v4l2-request-test.c (hunk 8 above)
Comment (rzerres) [reply v4l2-request-test.c]: ok. will throw it away.

File: v4l2-request-test.c (hunk 3 above)
Comment (rzerres) [reply v4l2-request-test.c]: done.

File: v4l2-request-test.c (hunk 4 above)
Comment (rzerres) [reply v4l2-request-test.c]: All discussed issues are incorporated in [v4]

File: v4l2-request-test.c (hunk 1 above)
Comment (paulkocialkowski) [reply v4l2-request-test.c]: Yes but this is done later in the code, so config.slices_path was not allocated yet. I would expect this to crash actually.

File: v4l2-request-test.c (hunk 2 above)
Comment (paulkocialkowski) [reply v4l2-request-test.c]: Alignment looks good now but you missed a few whitespaces before `},`

File: v4l2-request-test.c [hunk 9]
```diff
@@ -64,42 +64,44 @@ struct format_description formats[] = {
...
+		" -D, --drm-driver <name>   Use given DRM driver.\n"
+		" -s, --slices-path <path>  Use <path> to find stored video slices.\n"
+		" -S, --slices-format <slices format>\n"
+		"                           Regex/format describing filenames stored in the slices path.\n"
+		" -f, --fps <fps>           Display given number of frames per seconds.\n"
+		" -P, --preset-name <name>  Use given preset-name for video decoding.\n"
+		" -i, --interactive         Enable interactive mode.\n"
+		" -l, --loop                Loop preset frames.\n"
+		" -q, --quiet               Enable quiet mode.\n"
+		" -h, --help                This help message.\n\n");
```
Comment (paulkocialkowski) [inline v4l2-request-test.c]: We probably should switch to tabs for the block above too.

File: v4l2-request-test.c [hunk 10]
```diff
@@ -64,42 +64,44 @@ struct format_description formats[] = {
...
-	       " -P [video preset]              video preset to use\n"
-	       " -i                             enable interactive mode\n"
-	       " -l                             loop preset frames\n"
-	       " -q                             enable quiet mode\n"
-	       " -h                             help\n\n"
-	       "Video presets:\n");
+	printf("Usage: v4l2-request-test [OPTIONS]\n\n"
+		"Options:\n"
+		" -v, --device <dev>\n"
+		"     --video-device <dev>  Use device <dev> as the video device.\n"
```
Comment (paulkocialkowski) [inline v4l2-request-test.c]: Can we have `--video-device` on the first line with the helper text and `--device` on the second one instead?

**Review by paulkocialkowski — COMMENTED** [review]
Here is another batch of review. Patch `update header inclusion for H.264 and HEVC` can be dropped now. Thanks!

**Comment by rzerres** [comment]
You can close this track.
I will send a new pull request, hopefully the final [v5].

**Pull request closed without merging**